contracts/src/v0.8/vrf/dev/VRFCoordinatorV2_5.sol

```diff
@@ -569,15 +580,19 @@ contract VRFCoordinatorV2_5 is VRF, SubscriptionAPI, IVRFCoordinatorV2Plus {
    * @inheritdoc IVRFSubscriptionV2Plus
    */
   function pendingRequestExists(uint256 subId) public view override returns (bool) {
-    SubscriptionConfig memory subConfig = s_subscriptionConfigs[subId];
-    for (uint256 i = 0; i < subConfig.consumers.length; i++) {
-      for (uint256 j = 0; j < s_provingKeyHashes.length; j++) {
-        (uint256 reqId, ) = _computeRequestId(
-          s_provingKeyHashes[j],
-          subConfig.consumers[i],
-          subId,
-          s_consumers[subConfig.consumers[i]][subId]
-        );
+    address[] storage consumers = s_subscriptionConfigs[subId].consumers;
+    uint256 consumersLength = consumers.length;
+    if (consumersLength == 0) {
+      return false;
+    }
+    uint256 provingKeyHashesLength = s_provingKeyHashes.length;
+    if (provingKeyHashesLength == 0) {
```

Comment:
we can assume contract always has proving keys so i think we can remove this check?

Reply:
I removed this check and the gas increased. I'm not sure if it's because of the way our tests are set up. Maybe it's testing for cases where there are no proving keys.

Reply:
This method is called from "CancelSubscription" and cancel subscription tests do not set up proving keys. So yeah, I think it's because most test cases that we have don't set up proving keys unless it's testing request/fulfillment flow

Reply:
Can we assume that the hot path will always have at least 1 consumer and at least 1 proving key? If that's the case I will remove both checks here. Note that this function is called by `removeConsumer`. `cancelSubscription`, and `migrate`.

The proving key length check is to prevent `address consumer = consumers[i]` in line 593 from being called if the inner loop wouldn't even be executed and the consumers length check is to prevent accessing `s_provingKeyHashes` storage in line 588 when the outer loop wouldn't be executed.

Reply:
we can assume there is always at least one proving key hash. When `removeConsumer`, `cancelSubscription`, `migrate` are called, `consumers` is more likely to be present but not always, so I think removing both checks make sense. 

> Can we assume that the hot path will always have at least 1 consumer and at least 1 proving key?

Hot path (request/fulfillment) won't be calling this method right? so optimization here is scoped to the functions you mentioned: `removeConsumer`, `cancelSubscription`, and `migrate`.

Reply:
Yea, I meant the hot path of `removeConsumer`, `cancelSubscription`, and `migrate`, and NOT the hot path of VRF as a whole.